File: backend/go/controller/nytimes_controller.go
package controller

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/osucoding/simple-weather-app/service"
)

func SetupNyTimesController(engine *gin.Engine) {
	routerGroup := engine.Group("/api/v0/ny-times")

	routerGroup.GET("/news", getNews)
}

func getNews(context *gin.Context) {
	sectionValue := context.Query("sectionValue")
	if len(sectionValue) == 0 {
		context.JSON(http.StatusBadRequest, "Not a valid sectionValue")
		return
	}

	nyTimesResponse, err := service.GetNews(sectionValue)
	if err != nil {
		context.JSON(http.StatusBadRequest, "Error during API call to NYTimes")
		return
	}
	context.JSON(http.StatusOK, nyTimesResponse)
}

File: backend/python/controller/nytimes_controller.py
from flask import Blueprint, Response, request
import service.nytimes_service as nytimes_service

blueprint = Blueprint(
    'nytimes_controller',
    __name__,
    url_prefix='/api/v0/ny-times'
)

@blueprint.route('/news', methods=['GET'])
def get_news():
    section_value = request.args.get('sectionValue')
    response = nytimes_service.get_news(section_value)
    return Response(response.model_dump_json(), status=200, mimetype='application/json')
